### PR TITLE
Fix hacks on aten.item

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136214

Differential Revision: [D62893782](https://our.internmc.facebook.com/intern/diff/D62893782)

### CODE REVIEW

File: torch/export/exported_program.py [hunk 1]
```diff
@@ -226,7 +227,23 @@ def _override_composite_implicit_decomp(cia_ops_to_callable, safe=True):
...
+    When we are running through real tensors, we do need to dispatch to C++ as the CPU
+    key now will hit the Python CIA which we are overriding to return NotIMplemented. 
+    """
+    kernel = kwargs["kernel"]
+    del kwargs["kernel"]
+
+    def _safe_to_be_delegated_to_cpp_dispatcher(x):
+        if not isinstance(x, torch.Tensor):
+            return True 
+        return isinstance(x, torch.Tensor) and not isinstance(x, FakeTensor)
```
Comment (tugsbayasgalan) [inline torch/export/exported_program.py:241]: @bdhirsh Should we assert for meta tensor here as well?

File: test/export/test_export.py [hunk 2]
```diff
@@ -3887,6 +3887,73 @@ def forward(self, x):
...
+def forward(self, c_lifted_tensor_0, x):
+    lift_fresh_copy = torch.ops.aten.lift_fresh_copy.default(c_lifted_tensor_0);  c_lifted_tensor_0 = None
+    _assert_async = torch.ops.aten._assert_async.msg(lift_fresh_copy, 'Fail');  lift_fresh_copy = _assert_async = None
+    return (x,)"""
+            )
+        elif is_retracebility_test(self._testMethodName):
+            self.assertExpectedInline(
+                str(ep.graph_module.code).strip(), """\
+def forward(self, c_lifted_tensor_0, x):
+    clone = torch.ops.aten.clone.default(c_lifted_tensor_0);  c_lifted_tensor_0 = None
```
Comment (tugsbayasgalan) [inline test/export/test_export.py:3912]: This is an existing behaviour, seems like lift_fresh_copy gets traced into clone op which makes sense?

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62893782).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62893782).

File: torch/export/exported_program.py [hunk 3]
```diff
@@ -226,7 +227,28 @@ def _override_composite_implicit_decomp(cia_ops_to_callable, safe=True):
...
 
 def _special_op_to_preserve_cia(*args, **kwargs):
-    "This is an special marker that tells our infra that we shouldn't decompose this op"
+    """
+    This is an special marker that tells our infra that we shouldn't decompose this op.
+    When we are running through real tensors, we do need to dispatch to C++ as the CPU
+    key now will hit the Python CIA which we are overriding to return NotIMplemented.
+    """
+    kernel = kwargs["kernel"]
+    del kwargs["kernel"]
```
Comment (Skylion007) [inline torch/export/exported_program.py:236]:
```suggestion
    kernel = kwargs.pop("kernel")
```

**Pull request closed without merging**